src/App.js

```diff
@@ -45,6 +48,12 @@ const App = () => {
         const unregister = chrome.on('GLOBAL_FILTER_UPDATE', ({ data }) => {
             const [workloads, SIDs, tags] = chrome.mapGlobalFilter(data, true, true) ?? [null, null, null];
             dispatch(setGlobalFilter({ workloads, SIDs, tags }));
+            // Always append the global filter to the URL
+            const hashSearchParams = new URLSearchParams();
+            hashSearchParams.set('workloads', Object.keys(workloads).toString());
+            hashSearchParams.set('SIDs', SIDs.toString());
+            hashSearchParams.set('tags', tags.toString());
+            navigate({ ...location, hash: hashSearchParams.toString() });
```

Comment:
previously when I did `navigate("#workloads")` the whole url would change and the params would be lost, so changed it to this